### PR TITLE
IG-2156 [Nuclio] Add "Duplicate" action to functions

### PR DESCRIPTION
**Pull request by YarovoiVadym**
IG-2156 [Nuclio] Add "Duplicate" action to functions

### CODE REVIEW

File: src/nuclio/projects/project/functions/duplicate-function-dialog/duplicate-function-dialog.component.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        ctrl.duplicateFunction = duplicateFunction;
+        ctrl.inputValueCallback = inputValueCallback;
+        ctrl.onClose = onClose;
+
+        //
+        // Public methods
+        //
+
+        /**
+         *  Wrapper for `FunctionsService.duplicateFunction` method
```
Comment (LevkoKozak) [inline src/nuclio/projects/project/functions/duplicate-function-dialog/duplicate-function-dialog.component.js]: Remove blank space before Wrapper word

File: src/nuclio/projects/project/functions/duplicate-function-dialog/duplicate-function-dialog.component.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+         */
+        function duplicateFunction(event) {
+            if (angular.isUndefined(event) || event.keyCode === EventHelperService.ENTER) {
+                ctrl.nameTakenError = false;
+                $scope.duplicateFunctionForm.$submitted = true;
```
Comment (LevkoKozak) [inline src/nuclio/projects/project/functions/duplicate-function-dialog/duplicate-function-dialog.component.js]: $scope?

File: src/nuclio/projects/project/functions/duplicate-function-dialog/duplicate-function-dialog.less [hunk 3]
```diff
@@ -0,0 +1,20 @@
+.duplicate-function-dialog-wrapper {
+    .main-content {
+        .field-group {
+            display: flex;
+            flex-wrap: wrap;
+            width: 450px;
+
+            .field-label,
+            .field-input {
+                width: 100% !important;
```
Comment (LevkoKozak) [inline src/nuclio/projects/project/functions/duplicate-function-dialog/duplicate-function-dialog.less]: Please use cascades styles instead.

File: src/nuclio/projects/project/functions/duplicate-function-dialog/duplicate-function-dialog.less [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+            }
+
+            .field-input {
+                .error {
+                    top: 38px !important;
```
Comment (LevkoKozak) [inline src/nuclio/projects/project/functions/duplicate-function-dialog/duplicate-function-dialog.less]: Please use cascades styles instead.

File: src/nuclio/projects/project/functions/duplicate-function-dialog/duplicate-function-dialog.component.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        function inputValueCallback(newData) {
+            ctrl.newFunctionName = newData;
+        }
+
+        /**
+         * Closes dialog
+         * @param {Event} [event]
+         */
+        function onClose(event) {
+            if ((angular.isUndefined(event) || event.keyCode === EventHelperService.ENTER) && !ctrl.isLoadingState) {
```
Comment (mavdryk) [inline src/nuclio/projects/project/functions/duplicate-function-dialog/duplicate-function-dialog.component.js:70]: it's better to place !ctrl.isLoadingState to the beginning of the condition

File: src/nuclio/projects/project/functions/duplicate-function-dialog/duplicate-function-dialog.tpl.html [hunk 6]
```diff
@@ -0,0 +1,61 @@
+<div class="close-button igz-icon-close" data-ng-click="$ctrl.onClose()"></div>
+
+<div class="title">
+    <span>
```
Comment (LevkoKozak) [inline src/nuclio/projects/project/functions/duplicate-function-dialog/duplicate-function-dialog.tpl.html]: redundant <span>?

**Pull request closed without merging**